### PR TITLE
Added cache on graphql schema

### PR DESCRIPTION
**Pull request by MaxSouza**
- Added cache on GraphQl schema
- Added a condition to avoid requests when the customer isn't logged in.

Issue #10

### CODE REVIEW

File: Model/Resolver/Identity.php [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<?php declare(strict_types=1);
+
+namespace Discorgento\ProductLastOrder\Model\Resolver;
+
+use Magento\Framework\GraphQl\Query\Resolver\IdentityInterface;
+
+class Identity implements IdentityInterface
```
Comment (jonatanaxe) [inline Model/Resolver/Identity.php]: Isn't this name 'class Identity' too generic, or is this a pattern?

File: Model/Resolver/Identity.php [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<?php declare(strict_types=1);
```
Comment (jonatanaxe) [inline Model/Resolver/Identity.php]: You could add the Copyright block

```
/**
 * Copyright ©  All rights reserved.
 * See COPYING.txt for license details.
 */
```

File: view/frontend/web/js/component-msg-last-order.js [hunk 3]
```diff
@@ -35,30 +35,13 @@ define([
...
-                            }
-                        }
-                    `,
-                    variables: {
-                        customerId: self.customerId,
-                        productId: self.productId
-                    }
-                })
+            let urlGraphql = urlBuilder.build('graphql');
+            let queryGraphql = 'query CheckCustomerProductPurchase($customerId: String!, $productId: String!) { hasCustomerPurchasedProduct(customerId: $customerId, productId: $productId) { hasPurchased orderLink orderDate }}&variables={"customerId": "'+ self.customerId +'", "productId": "'+self.productId+'"}';
```
Comment (jonatanaxe) [inline view/frontend/web/js/component-msg-last-order.js]: Can we leave this query indented, for better reading? If not, could you tell me why?

File: Model/Resolver/Identity.php (hunk 1 above)
Comment (jonatanaxe) [reply Model/Resolver/Identity.php:7]: @MaxSouza

File: Model/Resolver/Identity.php (hunk 2 above)
Comment (jonatanaxe) [reply Model/Resolver/Identity.php]: @MaxSouza

File: view/frontend/web/js/component-msg-last-order.js (hunk 3 above)
Comment (jonatanaxe) [reply view/frontend/web/js/component-msg-last-order.js:41]: @MaxSouza

File: Model/Resolver/Identity.php (hunk 1 above)
Comment (MaxSouza) [reply Model/Resolver/Identity.php]: It's a pattern.

We could change that, but the suffix should keep. 

Example: SomethingIdentity
image

Comment (jonatanaxe) [reply Model/Resolver/Identity.php]: So we can put something like this `ProductLastOrderIdentity` ? would it be right?

Comment (MaxSouza) [reply Model/Resolver/Identity.php]: Yeap, it would be

File: view/frontend/web/js/component-msg-last-order.js (hunk 3 above)
Comment (MaxSouza) [reply view/frontend/web/js/component-msg-last-order.js]: Yes, we can leave, but I've adjusted it.

File: Model/Resolver/Identity.php (hunk 1 above)
Comment (jonatanaxe) [reply Model/Resolver/Identity.php]: Cool good to know, you could use this for us

Comment (MaxSouza) [reply Model/Resolver/Identity.php]: Done

**Review by maria-axe — APPROVED** [review]
Tests performed successfully in Development and Production environments.

**Review by jonatanaxe — APPROVED** [review]

**Pull request merged**